### ISSUE TITLE
Break statement not handled properly in code flow logic

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#192]
See bug report that @giyyapan appended to https://github.com/microsoft/pyright/issues/189

**Comment by RaptDept** [comment]
I'm having a slightly different, but closely related, issue with `break` inside `for` or `while`, and I'm not sure if it warrants its own issue so I'm posting in this thread first.

In this code, everything after the first line inside the `for` or `while` is seen as unreachable.
```
for _ in []: # also applies to "while True:"
  print('Everything after this line is unreachable')
  print('Unreachable 1')
  print('Unreachable 2')
  print('Unreachable 3')
  break # also unreachable
```

This can be annoying for code that looks like this
```
for element in some_list:
  if some_condition:
    continue

  if other_condition:
    continue

  # more condition checking...

  # found first element that passes all checks, do processing here
  break
```

**Comment by erictraut** [comment]
@RaptDept, thanks for that bug report. I'm going to create a new issue to track this, since it's different from the original bug.

**Comment by erictraut** [comment]
Thanks again for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is now fixed in 1.0.58, which I just published.
